controllers/applicationvolumereplication_controller.go

```diff
@@ -407,7 +445,9 @@ func (r *ApplicationVolumeReplicationReconciler) vrgManifestWorkAlreadyExists(
 
 	if d, found := avr.Status.Decisions[subscription.Name]; found {
```

Comment:
Should we be certain the status that we are using belongs to this generation of the CR? The MW is created or updated later, so functionally that would not be a problem. The Status update though is skipped or remains as is, hence the question.

Reply:
Subscription is either paused or unpause.  If it is unpaused, then the status.Decision either does not contain the subscription (in the case of the initial deployment) OR it does contain the subscription but the MW for it, is guaranteed to be deleted due to transitioning from a paused subscription to unpaused subscription. Nonetheless, in my latest changes, I am adding another layer of protection, which will be in a different PR.